Dockerfile

```diff
@@ -1,31 +1,59 @@
-FROM php:7.2-fpm
+ARG COMPOSER_IMAGE_VERSION=1.10
+ARG NODE_IMAGE_VERSION=14.4
+ARG PHP_IMAGE_VERSION=7.2-fpm-buster
 
-# Install system packages (Git, packages for archives, Node.js)
-RUN apt-get update \
-    && apt-get install -y git libzip-dev zlib1g-dev unzip \
-    && curl -sL https://deb.nodesource.com/setup_14.x | bash - && apt-get install -y nodejs \
-    && rm -rf /var/lib/apt/lists/*
+FROM composer:${COMPOSER_IMAGE_VERSION} as composer
 
-# Install PHP extensions
-RUN docker-php-ext-install pdo_mysql && docker-php-ext-install zip && docker-php-ext-install calendar
+FROM node:${NODE_IMAGE_VERSION} as assets
+
+WORKDIR /build
+
+COPY yarn.lock package.json webpack.mix.js ./
+
+RUN yarn install
+
+COPY ./resources ./resources
+
+RUN yarn run production
 
-# Ensure commands are running successfully before continuing
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+FROM php:${PHP_IMAGE_VERSION}
 
-# Install Composer
-RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+ARG NGINX_PACKAGE_VERSION
+ARG SUPERVISOR_PACKAGE_VERSION
+ARG CRON_PACKAGE_VERSION
 
-# Install Yarn
-RUN npm install -g yarn
+# Install nginx as webserver, and some packages needed for php extensions
+RUN apt-get update && apt-get install -y \
+    nginx-light \
+    supervisor \
+    cron \
+    libzip-dev \
+    unzip \
+ && rm -rf /var/lib/apt/lists/*
+
+# Install PHP extensions
+RUN docker-php-ext-install pdo_mysql \
+ && docker-php-ext-install zip \
+ && docker-php-ext-install calendar
```

Comment:
No big deal, but can make sure we use the same level of indentation here as on lines 27, 28, etc?